Source/Core/Xml.cs

```diff
@@ -114,36 +114,26 @@ public void WriteEndMethod(string outcome, DateTime endTime, TimeSpan elapsed, i
       cce.EndExpose();
     }
 
-    public void WriteStartSplit(int splitNum, DateTime startTime)
+    public void WriteSplit(int splitNum, DateTime startTime, string outcome, TimeSpan elapsed)
     {
       Contract.Requires(splitNum > 0);
+      Contract.Requires(outcome != null);
       Contract.Requires(IsOpen);
       //modifies this.*;
       Contract.Ensures(IsOpen);
       Contract.Assert(wr != null);
+
       cce.BeginExpose(this);
-      {
+      lock (this) {
```

Comment:
Is there any point in adding locking in the `XmlSink` class itself, given you have to lock around the sequence of multiple calls on the caller side anyway?

Reply:
Perhaps not. I added them before realizing that there would still need to be a top-level lock. I'll experiment.